lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -529,9 +553,14 @@ def datasets(
                         raise APIAbort(
                             HTTPStatus.BAD_REQUEST, str(MetadataBadKey(k))
                         ) from e
-                    sorter = order(c)
+
+                    # For native SQL columns, use the SQL type unless
+                    # explicitly overridden.
+                    sorter = order(c if defaulted_type else c.cast(cast_to))
             if sorter is None:
-                sorter = order(aliases[native_key].value[keys])
+                casted = cast(aliases[native_key].value[keys].as_string(), cast_to)
+                query = query.add_column(casted)
```

Comment:
@dbutenhof, I just noticed that we're getting a deprecation warning on this statement:
```
...
test_datasets_list.py::TestDatasetsList::test_dataset_sort[user.test.odd,global.test.sequence:desc-results7]
  /var/tmp/jenkins/tox/py39/lib/python3.9/site-packages/pbench/server/api/resources/datasets_list.py:562: SADeprecationWarning: Query.add_column() is deprecated and will be removed in a future release.  Please use Query.add_columns() (deprecated since: 1.4)
    query = query.add_column(casted)
```